src/amo/utils/index.js

```diff
@@ -42,25 +42,32 @@ export function getAddonURL(slug: string): string {
   return `/addon/${slug}/`;
 }
 
+// Note that this also adds `experiment` and `variation` if provided.
```

Comment:
I think we shouldn't have to add this comment. Instead, we could allow any other query parameter. The function is named `makeQueryStringWithUTM()` but it uses `makeQueryString()` under the hood so I think it's fine to pass extra query params.

I suggest we allow `makeQueryStringWithUTM()` to have any query parameters in addition to the 5 `utm_*` ones. Maybe we could write

```js
type MakeQueryStringWithUtmParams = {|
  utm_campaign?: string | null,
  utm_content?: string | null,
  utm_medium?: string,
  utm_source?: string,
  utm_term?: string | null,
  otherQueryParams?: { [name: string]: string | null },
|};

export const makeQueryStringWithUTM = ({
  utm_campaign = DOWNLOAD_FIREFOX_UTM_CAMPAIGN,
  utm_content = null,
  utm_medium = DEFAULT_UTM_MEDIUM,
  utm_source = DEFAULT_UTM_SOURCE,
  utm_term = null,
  ...otherQueryParams = {}
}: MakeQueryStringWithUtmParams): string => { ... }
```

Reply:
Good idea, I am updating it.